cpp/tests/CMakeLists.txt

```diff
@@ -65,29 +65,29 @@ include_directories("${CMAKE_BINARY_DIR}/include"
                     "${CMAKE_SOURCE_DIR}/include"
                     "${CMAKE_SOURCE_DIR}"
                     "${CMAKE_SOURCE_DIR}/src"
-                    "${CMAKE_SOURCE_DIR}/../thirdparty/cub"
-                    "${CMAKE_SOURCE_DIR}/../thirdparty/libcudacxx/include"
-                    "${CMAKE_SOURCE_DIR}/thirdparty/dlpack/include"
+                    "${GDAL_INCLUDE}"
                     "${GTEST_INCLUDE_DIR}"
                     "${RMM_INCLUDE}"
                     "${CUDF_INCLUDE}"
                     "${CUDF_SRC_INCLUDE}"
                     "${CUDF_TEST_INCLUDE}"
-                    "${CUB_INCLUDE}"
+                    "${CUB_INCLUDE_DIR}"
```

Comment:
CUB and Thrust should be included before the CUDA toolkit include path to avoid getting the CUDA toolkit version of them. See how this is done in libcudf cmake.